src/RingGrids/geodesics.jl

```diff
@@ -0,0 +1,54 @@
+const DEFAULT_RADIUS = 6371e3  # Earth mean radius in meters.
+
+"""
+    abstract type AbstractSphericalDistance end
+
+Super type of formulas to calculate the spherical distance or great-circle distance.
+To define a `NewFormula`, define `struct NewFormula <: AbstractSphericalDistance end`
+and the actual calculation as a functor
+
+    function NewFormula(lonlat1::Tuple, lonlat2::Tuple; radius=DEFAULT_RADIUS, kwargs...)
+
+assuming inputs in degrees and returning the distance in meters (or radians for radius=1).
+Then use the general interface `spherical_distance(NewFormula, args...; kwargs...)`"""
+abstract type AbstractSphericalDistance end
+struct Haversine <: AbstractSphericalDistance end
+
+"""$(TYPEDSIGNATURES) 
+Haversine formula calculating the great-circle or spherical distance (in meters) on the sphere
+between two tuples of  longitude-latitude points in degrees ˚E, ˚N. Use keyword argument `radius`
+to change the radius of the sphere (default 6371e3 meters, Earth's radius), use `radius=1`
+to return the central angle in radians."""
+function Haversine(             # functor for Haversine struct
+    lonlat1::Tuple,             # point 1 in spherical coordinates
+    lonlat2::Tuple;             # point 2
+    radius = DEFAULT_RADIUS,    # radius of the sphere [m]
+)
+    lon1, lat1 = lonlat1
+    lon2, lat2 = lonlat2    
+
+    φ1 = deg2rad(lat1)
+    φ2 = deg2rad(lat2)
+
+    Δφ = deg2rad(lat2 - lat1)
+    Δλ = deg2rad(lon2 - lon1)
+
+    # Haversine formula, see https://en.wikipedia.org/wiki/Haversine_formula
+    a = sin(Δφ / 2)^2 + cos(φ1) * cos(φ2) * sin(Δλ / 2)^2
+    c = 2 * atan(sqrt(a), sqrt(1 - a))
+    return c*radius             # in meters
```

Comment:
There is one more issue here: `radius::Float64` by default so it's always float64 that is returned, but it would be nice to have the function being type stable, in a sense of that the promotion of the input types in the tuples determines the output type. We could either do `c*convert(typeof(c), radius)` but then a user defined radius would be inferior to the latlons or we simply define the `const DEFAULT_RADIUS = 6371000` as integer which is probably my preferred option!